bundle-workflow/src/run_integ_test.py

```diff
@@ -59,25 +59,15 @@ def main():
                     work_dir,
                     args.s3_bucket
                 )
-                test_configs = test_suite.execute()
-                for security, status in test_configs.items():
-                    if status != 0:
-                        failed_components[(component.name, security)] = status
-                    else:
-                        passed_components[(component.name, security)] = status
+                test_suite.execute(test_result)
```

Comment:
Instead of test_configs, the `execute()` should return a list of `test_result` for each tested configs. This `test_result` is an instance of `TestResult` class, with properties as mentioned [here](https://github.com/opensearch-project/opensearch-build/pull/641/files#r718559457) and the `TestResults` should be a collection of `TestResult` and provides a summary method. @dblock 's comment sums the requirements nicely.